internal/settings/config_provider.go

```diff
@@ -83,41 +83,88 @@ func (c *configProvider) GetExcludePrefixesForDir(dirPath string) ([]string, err
 	return getExcludePrefixesForDir(dirPath)
 }
 
-// getFilePathForDir tries to find a file named DefaultConfigFilename starting in the
+// getFilePathForDir tries to find a file named by one of the ConfigFilenames starting in the
 // given directory, and going up a directory until hitting root.
 //
 // The directory must be an absolute path.
 //
 // If no such file is found, "" is returned.
-// Also returns all the directories this Config applies to.
-func getFilePathForDir(dirPath string) (string, []string) {
-	var dirPaths []string
+// If multiple files named by one of the ConfigFilenames are found in the same
+// directory, error is returned.
+func getFilePathForDir(dirPath string) (string, error) {
 	for {
-		dirPaths = append(dirPaths, dirPath)
-		filePath := filepath.Join(dirPath, DefaultConfigFilename)
-		if _, err := os.Stat(filePath); err == nil {
-			return filePath, dirPaths
+		filePath, err := getSingleFilePathForDir(dirPath)
+		if err != nil {
+			return "", err
+		}
+		if filePath != "" {
+			return filePath, nil
 		}
 		if dirPath == "/" {
-			return "", dirPaths
+			return "", nil
 		}
 		dirPath = filepath.Dir(dirPath)
 	}
 }
 
+// getSingleFilePathForDir gets the file named by one of the ConfigFilenames in the
+// given directory. Having multiple such files results in an error being returned. If no file is
+// found, this returns "".
+func getSingleFilePathForDir(dirPath string) (string, error) {
```

Comment:
nit: can we just call this `getConfigPath` or similar? Using `single` seems a bit odd since we will only ever expect to have one file path for a configuration.

Reply:
This is matching the above `getFilePathForDir`, which performs the logic to go up to root for the given directory path. I think having `getFilePathForDir` and `getConfigPath` confuses the purpose of these two functions.